skimage/feature/util.py

```diff
@@ -145,16 +146,29 @@ def plot_matched_features(
     ax.imshow(image, cmap='gray')
     ax.axis((0, image0.shape[1] + offset[1], image0.shape[0] + offset[0], 0))
 
-    rng = np.random.default_rng()
+    rng = np.random.default_rng(seed=0)
 
-    for i in range(matches.shape[0]):
-        idx0 = matches[i, 0]
-        idx1 = matches[i, 1]
+    number_of_matches = matches.shape[0]
 
-        if matches_color is None:
-            color = rng.random(3)
-        else:
-            color = matches_color
+    from matplotlib.colors import is_color_like
+
+    if matches_color is None:
+        colors = [rng.random(3) for _ in range(number_of_matches)]
+    elif is_color_like(matches_color):
+        colors = [matches_color for _ in range(number_of_matches)]
+    elif isinstance(matches_color, Sequence):
+        if len(matches_color) != number_of_matches:
+            raise ValueError(
+                '``matches_color`` needs to be a single color or a sequence with a color for each match.'
+            )
+        colors = matches_color
+    else:
+        raise ValueError(
+            '``matches_color`` needs to be a single color or a sequence with a color for each match.'
+        )
```

Comment:
Nitpick: this error message is repeated, could you pass it as a variable to `ValueError`? This decreases the traceback verbosity as well as a minor side effect. 

Also, I think we usually indicate parameters with a single backtick character instead of two.

Reply:
Good point. I introduced the variable and only used the single backtick parameter.